src/cogent3/core/location.py

```diff
@@ -1077,6 +1079,101 @@ def _input_vals_neg_step(
     return (0, 0, 1) if start < stop else (start, stop, step)
 
 
+@numba.jit
+def seq_to_align_index(
+    gap_pos: IntArrayTypes,
+    cum_lengths: IntArrayTypes,
+    parent_length: int,
```

Comment:
just a comment that in general we should document that the parent is the seq, sometimes I get confused which length parent_length refers to (alignment or seq)